tests/common/chain_mock.rs

```diff
@@ -126,31 +130,24 @@ pub async fn spawn_mock_chain(
         .clone()
         .deploy((first_batch_size as u64, mock_verifier.address()))?
         .confirmations(0usize)
-        .send();
+        .send()
+        .await?;
 
     let update_verifiers = verifier_lookup_table_factory
         .deploy((first_batch_size as u64, unimplemented_verifier.address()))?
         .confirmations(0usize)
-        .send();
+        .send()
+        .await?;
 
     let identity_manager_impl_factory =
         load_and_build_contract("./sol/WorldIDIdentityManagerImplV1.json", client.clone())?;
 
     let identity_manager_impl = identity_manager_impl_factory
         .deploy(())?
         .confirmations(0usize)
-        .send();
-
-    let (insert_verifiers, update_verifiers, identity_manager_impl) =
-        tokio::join!(insert_verifiers, update_verifiers, identity_manager_impl);
-
-    let insert_verifiers = insert_verifiers?;
-    let update_verifiers = update_verifiers?;
-    let identity_manager_impl = identity_manager_impl?;
+        .send()
+        .await?;
 
-    // TODO: This is sequential but could be parallelized.
```

Comment:
this TODO should stay? The new impl looks very sequential?

Reply:
Yes, but that was the issue. Parallelizing it caused nonce issues. It's safer to just run everything sequentially until we have a good idea on how to handle that